pkg/conversion/conversion.go

```diff
@@ -0,0 +1,124 @@
+			Replicas:         assignedIdentity.Spec.Replicas,
+		},
+		Status: aadpodid.AzureAssignedIdentityStatus(assignedIdentity.Status),
+	}
+}
+
+func ConvertInternalBindingToV1Binding(identityBinding aadpodid.AzureIdentityBinding) (resIdentityBinding aadpodidv1.AzureIdentityBinding) {
+	return aadpodidv1.AzureIdentityBinding{
+		TypeMeta:   identityBinding.TypeMeta,
+		ObjectMeta: identityBinding.ObjectMeta,
+		Spec: aadpodidv1.AzureIdentityBindingSpec{
+			ObjectMeta:    identityBinding.Spec.ObjectMeta,
+			AzureIdentity: identityBinding.Spec.AzureIdentity,
+			Selector:      identityBinding.Spec.Selector,
+			Weight:        identityBinding.Spec.Weight,
+		},
+		Status: aadpodidv1.AzureIdentityBindingStatus(identityBinding.Status),
+	}
+}
+
+func ConvertInternalIdentityToV1Identity(identity aadpodid.AzureIdentity) (resIdentity aadpodidv1.AzureIdentity) {
+	return aadpodidv1.AzureIdentity{
+		TypeMeta:   identity.TypeMeta,
+		ObjectMeta: identity.ObjectMeta,
+		Spec: aadpodidv1.AzureIdentitySpec{
+			ObjectMeta:     identity.Spec.ObjectMeta,
+			Type:           aadpodidv1.IdentityType(identity.Spec.Type),
+			ResourceID:     identity.Spec.ResourceID,
+			ClientID:       identity.Spec.ClientID,
+			ClientPassword: identity.Spec.ClientPassword,
+			TenantID:       identity.Spec.TenantID,
+			ADResourceID:   identity.Spec.ADResourceID,
+			ADEndpoint:     identity.Spec.ADEndpoint,
+			Replicas:       identity.Spec.Replicas,
+		},
+		Status: aadpodidv1.AzureIdentityStatus(identity.Status),
+	}
+}
+
+func ConvertInternalAssignedIdentityToV1AssignedIdentity(assignedIdentity aadpodid.AzureAssignedIdentity) (resAssignedIdentity aadpodidv1.AzureAssignedIdentity) {
+	retIdentity := ConvertInternalIdentityToV1Identity(*assignedIdentity.Spec.AzureIdentityRef)
+	retBinding := ConvertInternalBindingToV1Binding(*assignedIdentity.Spec.AzureBindingRef)
+
+	return aadpodidv1.AzureAssignedIdentity{
+		TypeMeta:   assignedIdentity.TypeMeta,
+		ObjectMeta: assignedIdentity.ObjectMeta,
+		Spec: aadpodidv1.AzureAssignedIdentitySpec{
+			ObjectMeta:       assignedIdentity.Spec.ObjectMeta,
+			AzureIdentityRef: &retIdentity,
+			AzureBindingRef:  &retBinding,
+			Pod:              assignedIdentity.Spec.Pod,
+			PodNamespace:     assignedIdentity.Spec.PodNamespace,
+			NodeName:         assignedIdentity.Spec.NodeName,
+			Replicas:         assignedIdentity.Spec.Replicas,
+		},
+		Status: aadpodidv1.AzureAssignedIdentityStatus(assignedIdentity.Status),
+	}
+}
+
+func ConvertV1AzurePodIdentityExceptionToInternalAzurePodIdentityException(idException aadpodidv1.AzurePodIdentityException) (residException aadpodid.AzurePodIdentityException) {
+	return aadpodid.AzurePodIdentityException{
+		TypeMeta:   idException.TypeMeta,
+		ObjectMeta: idException.ObjectMeta,
+		Spec: aadpodid.AzurePodIdentityExceptionSpec{
+			ObjectMeta: idException.Spec.ObjectMeta,
+			PodLabels:  idException.Spec.PodLabels,
+		},
+		Status: aadpodid.AzurePodIdentityExceptionStatus(idException.Status),
```

Comment:
Don't we need an internal Exception to v1 exception ? Is it because we don't create the exception list programatically ? Could you please add a comment to indicate that ?

Reply:
Yes - we only grab the list of exceptions and operate on them internally, but they are not created by the project, which is why I did not create the internal -> v1 function. Will add a comment